base/scripts/code.observer.cs

```diff
@@ -1,231 +1,335 @@
-      }
-   }
-   else
-   {
-      Client::limitCommandBandwidth(%clientId, false);
-      if(%clientId.observerMode == "commander")
-      {
-         Client::setControlObject(%clientId, %clientId.lastControlObject);
-		   %clientId.lastControlObject = "";
-         %clientId.observerMode = "";
-         %clientId.observerTarget = "";
-	   }
-   }
-}
\ No newline at end of file
+$GuiModeCommand    = 2;
+$LastControlObject = 0;
+
+function Observer::triggerDown(%client)
+{
+}
+
+function Observer::orbitObjectDeleted(%cl)
+{
+}
+
+function Observer::leaveMissionArea(%cl)
+{
+}
+
+function Observer::enterMissionArea(%cl)
+{
+}
+
+function Observer::triggerUp(%client)
+{
+   if(%client.observerMode == "dead")
+   {
+      if(%client.dieTime + $Server::respawnTime < getSimTime())
+      {
+         if(Game::playerSpawn(%client, true))
+         {
+            %client.observerMode = "";
+            Observer::checkObserved(%client);
+         }
+      }
+   }
+   else if ((%client.observerMode == "observerOrbit") || (%client.observerMode == "observerFirst"))
+      Observer::nextObservable(%client);
+   else if (%client.observerMode == "observerFlag")
+      Observer::nextFlagObs(%client);
+   else if(%client.observerMode == "observerFly")
+   {
+      %camSpawn = Game::pickObserverSpawn(%client);
+      Observer::setFlyMode(%client, GameBase::getPosition(%camSpawn), 
+          GameBase::getRotation(%camSpawn), true, true);
+   }
+   else if(%client.observerMode == "justJoined")
+   {
+      %client.observerMode = "";
+      Game::playerSpawn(%client, false);
+   }
+   else if(%client.observerMode == "pregame" && $Server::TourneyMode)
+   {
+      if($CountdownStarted)
+         return;
+
+      if(%client.notready)
+      {
+         %client.notready = "";
+         MessageAll(0, Client::getName(%client) @ " is READY.");
+         if(%client.notreadyCount < 3)
+            bottomprint(%client, "<f1><jc>Waiting for match start (FIRE if not ready).", 0);
+         else 
+            bottomprint(%client, "<f1><jc>Waiting for match start.", 0);
+      }
+      else
+      {
+         %client.notreadyCount++;
+         if(%client.notreadyCount < 4)
+         {
+            %client.notready = true;
+            MessageAll(0, Client::getName(%client) @ " is NOT READY.");
+            bottomprint(%client, "<f1><jc>Press FIRE when ready.", 0);
+         }
+         return;
+      }
+      Game::CheckTourneyMatchStart();
+   }
+}
+function Observer::CheckFlagStatus(%flagTeam)
+{
+    
+    for(%i=0; %i<$Observer::maxObservers; %i++) {
+        
+        if($ObservingFlag[ $ObserverClient[%i] ] == %flagTeam) {
+    
+            if($freeze::FlagClient[%flagTeam] == 0) {
+            
+                Observer::setTargetObject($ObserverClient[%i], $teamFlag[%flagTeam]);
+        
+            }
+            else {
+            
+                Observer::setTargetObject($ObserverClient[%i], $freeze::FlagClient[%flagTeam]);
+            }

```

Comment:
seeking to learn - if the flag is being carried what's the difference between setting the observer target object to the client of the flag carrier versus the flag itself?